### PR TITLE
s390x jenkinsfile

### PR DESCRIPTION
**Pull request by Nayana-ibm**
Adding jenkinsfile to run jenkins job on s390x node using jdk11 
We have removed camel-kudu and docs module as these are not supported on s390x.

### CODE REVIEW

**Comment by orpiske** [comment]
> 

Hello, thanks for contributing this. I do have a question, though: is this for running on s390 at Apache? I don't see any s390 node there.

**Comment by oscerd** [comment]
As far as I know we don't have s390 instances. So if this is used in a fork, it must be there. Upstream on ASF it makes less sense

**Comment by orpiske** [comment]
> As far as I know we don't have s390 instances. So if this is used in a fork, it must be there. Upstream on ASF it makes less sense

+1. 

In the case you want to include that node into the Apache CI (which, IMHO, I would love that), so that we have a s390x slave, you probably need to create a ticket to the Apache Infra. Here's a sample ticket you can look at [1] (that was for adding ppc64le nodes). If you already have a ticket, it would be good to link it here. 

Thanks!


1. https://issues.apache.org/jira/browse/INFRA-22612

**Comment by Nayana-ibm** [comment]
> > As far as I know we don't have s390 instances. So if this is used in a fork, it must be there. Upstream on ASF it makes less sense
> 
> +1.
> 
> @Nayana-ibm In the case you want to include that node into the Apache CI (which, IMHO, I would love that), so that we have a s390x slave, you probably need to create a ticket to the Apache Infra. Here's a sample ticket you can look at [1] (that was for adding ppc64le nodes for Kafka, but the process would be similar for Camel). If you already have a ticket, it would be good to link it here.
> 
> Thanks!
> 
>     1. https://issues.apache.org/jira/browse/INFRA-22612

We have already added s390x nodes to Apache CI 
listed here:  https://jenkins-ccos.apache.org/view/all/  (jenkins-s390x-1, jenkins-s390x-2, jenkins-s390x-3, jenkins-s390x-4)
issue: https://issues.apache.org/jira/browse/INFRA-21433

**Comment by orpiske** [comment]
> > > As far as I know we don't have s390 instances. So if this is used in a fork, it must be there. Upstream on ASF it makes less sense
> > 
> > 
> > +1.
> > @Nayana-ibm In the case you want to include that node into the Apache CI (which, IMHO, I would love that), so that we have a s390x slave, you probably need to create a ticket to the Apache Infra. Here's a sample ticket you can look at [1] (that was for adding ppc64le nodes for Kafka, but the process would be similar for Camel). If you already have a ticket, it would be good to link it here.
> > Thanks!
> > ```
> > 1. https://issues.apache.org/jira/browse/INFRA-22612
> > ```
> 
> We have already added s390x nodes to Apache CI listed here: https://jenkins-ccos.apache.org/view/all/ (jenkins-s390x-1, jenkins-s390x-2, jenkins-s390x-3, jenkins-s390x-4) issue: https://issues.apache.org/jira/browse/INFRA-21433

Thank you for clarifying this! In this case, it's +1 from me.

File: Jenkinsfile.s390x [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+        stage('Clean workspace') {
+             when {
+                 expression { params.CLEAN }
+             }
+             steps {
+                 sh 'git clean -fdx'
+                 // Remove camel-kudu and docs as these are not supported on s390x.
+                 sed -i '/camel-kudu/d' components/pom.xml
+                 sed -i '0,/docs/{//d;}' pom.xml
```
Comment (orpiske) [inline Jenkinsfile.s390x:60]: Maybe it's easier to run: 
`mvn -pl "!camel-kudu,!docs" -P install` 

Are you able to check if that works?

**Review by orpiske — APPROVED** [review]
+1 from me, but I did add a note for checking if can avoid some part of the CI logic by skipping incompatible modules via Maven CLI.

File: Jenkinsfile.s390x (hunk 1 above)
Comment (Nayana-ibm) [reply Jenkinsfile.s390x:60]: yes. below command worked:
mvn -Dmaven.test.skip.exec=true clean install -pl \!docs,\!components/camel-kudu

I will make changes to file

Comment (Nayana-ibm) [reply Jenkinsfile.s390x]: made changes. please review

Comment (orpiske) [reply Jenkinsfile.s390x]: Looking good, thanks.

**Comment by Nayana-ibm** [comment]
Thanks @orpiske 
Just wondering when s390x job will be executed on https://builds.apache.org/job/Camel/
Anything else needs to be done?

**Comment by orpiske** [comment]
> Thanks @orpiske Just wondering when s390x job will be executed on https://builds.apache.org/job/Camel/ Anything else needs to be done?

I think I need to work with @oscerd to include that job on the Apache CI before the job is available. I'll check this today.

**Comment by orpiske** [comment]
@Nayana-ibm actually, I could just copy/edit the CI configuration and added the job. You can watch it here: https://ci-builds.apache.org/job/Camel/job/Camel%20JDK11%20Build%20(S390x)/job/main/6/console

**Comment by orpiske** [comment]
@Nayana-ibm FYI, the build seems stuck waiting for the s390x nodes to come online.

**Comment by Nayana-ibm** [comment]
@orpiske Yes..nodes are offline right now. I will check with @gmcdonald

**Comment by Nayana-ibm** [comment]
@orpiske nodes are online now.  
Build fails with an error: _Error: JAVA_HOME is not defined correctly._
Do I need to update jenkinsfile to set PATH or jenkins job configuration will handle this?

**Comment by orpiske** [comment]
> @orpiske nodes are online now. Build fails with an error: _Error: JAVA_HOME is not defined correctly._ Do I need to update jenkinsfile to set PATH or jenkins job configuration will handle this?

I believe the JAVA_HOME has to be configured at the node configuration, no? If not, I think it will need to be set on the Jenkinsfile, indeed.

**Comment by Nayana-ibm** [comment]
> > @orpiske nodes are online now. Build fails with an error: _Error: JAVA_HOME is not defined correctly._ Do I need to update jenkinsfile to set PATH or jenkins job configuration will handle this?
> 
> I believe the JAVA_HOME has to be configured at the node configuration, no? If not, I think it will need to be set on the Jenkinsfile, indeed.

I think it should be set on node configuration.  As I don't see any other camel jenkinsfile sets the JAVA_HOME

**Comment by Nayana-ibm** [comment]
@orpiske Can you please trigger now?

**Comment by orpiske** [comment]
> @orpiske Can you please trigger build now?

Failed with: 

```
Error: JAVA_HOME is not defined correctly.
  We cannot execute /home/jenkins/tools/java/latest11/bin/java
```

**Comment by orpiske** [comment]
@Nayana-ibm BTW, if you want, feel free to ping me on Zulip chat and we can try to sort it out quicker. You can find me at https://camel.zulipchat.com/.

**Comment by gmcdonald** [comment]
Looks like there is a bit more work to do on these nodes, taking a look now

**Comment by Nayana-ibm** [comment]
issue is resolved now. Builds are executing https://builds.apache.org/job/Camel/job/Camel%20JDK11%20Build%20(S390x)/

**Pull request merged**